cpp/dolfinx/fem/CoordinateElement.cpp

```diff
@@ -5,30 +5,57 @@
 // SPDX-License-Identifier:    LGPL-3.0-or-later
 
 #include "CoordinateElement.h"
+#include <basix.h>
 #include <unsupported/Eigen/CXX11/Tensor>
 
 using namespace dolfinx;
 using namespace dolfinx::fem;
 
 //-----------------------------------------------------------------------------
 CoordinateElement::CoordinateElement(
-    mesh::CellType cell_type, int topological_dimension,
-    int geometric_dimension, const std::string& signature,
-    const ElementDofLayout& dof_layout, bool is_affine,
-    const std::function<int(double*, int, int, const double*)>&
-        evaluate_basis_derivatives)
-    : _tdim(topological_dimension), _gdim(geometric_dimension),
-      _cell(cell_type), _signature(signature), _dof_layout(dof_layout),
-      _is_affine(is_affine),
-      _evaluate_basis_derivatives(evaluate_basis_derivatives)
+    int basix_element_handle, int geometric_dimension,
+    const std::string& signature, const ElementDofLayout& dof_layout,
+    bool needs_permutation_data,
+    std::function<int(int*, const uint32_t)> permute_dofs,
+    std::function<int(int*, const uint32_t)> unpermute_dofs)
+    : _gdim(geometric_dimension), _signature(signature),
+      _dof_layout(dof_layout), _basix_element_handle(basix_element_handle),
+      _needs_permutation_data(needs_permutation_data),
+      _permute_dofs(permute_dofs), _unpermute_dofs(unpermute_dofs)
 {
+  const mesh::CellType cell = cell_shape();
+  int degree = basix::degree(basix_element_handle);
+  _is_affine
```

Comment:
We do export `is_affine` in ufc.h, could it be reused? Preferably both should come from basix alone?

Reply:
I've added this to issue #1338